### ISSUE TITLE
api_generator generates not all enums

### ISSUE THREAD

**Issue by maxim-sheronov** [godotengine/godot#11183]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Godot 3.0, master branch

**Issue description:**
You should run engine with --gdnative-generate-json-api flag
In output api.json file you can see, that type GeometryInstance does not have any enums
In engine sources in file visual_instance.h declared two enums Flags and ShadowCastingSetting

api_generator.cpp tries to get enums from ClassDB::get_enum_list, but it does not contain these enums

**Types with this issue:**
AnimationTreePlayer, Area, Area2D, Mesh, AudioEffectFilter, AudioEffectDistortion, AudioServer, AudioStreamPlayer, AudioStreamSample, CollisionPolygon2D, Curve, EditorFileDialog, ImageTexture, Light2D

**Additional info:**
There is one more problem.
enums LineTextureMode, LineCapMode, LineJointMode defined in namespace, but not in class Line2D. But somehow in api.json they appears in Line2D class, but in other places, where is usage of these enums they looks like enum.LineTextureMode and not like enum.Line2D::LineTextureMode

**Comment by maxim-sheronov** [comment]
One more example:
AnimationTreePlayer has only one enum NodeType and not has AnimationProcessMode

**Comment by akien-mga** [comment]
CC @neikeq

**Comment by maxim-sheronov** [comment]
More types with this issue:
Area, Area2D, Mesh

**Comment by neikeq** [comment]
Should use `BIND_ENUM_CONSTANT` instead of `BIND_CONSTANT`. If we upgrade to C++11 an error or warning about this would be possible.

**Comment by maxim-sheronov** [comment]
There is one more problem.
enums LineTextureMode, LineCapMode, LineJointMode defined in namespace, but not in class Line2D. But somehow in api.json they appears in Line2D class, but in other places, where is usage of these enums they looks like enum.LineTextureMode and not like enum.Line2D::LineTextureMode

**Comment by maxim-sheronov** [comment]
@neikeq I see only usages of VARIANT_ENUM_CAST macro

**Comment by neikeq** [comment]
They are in `_bind_methods()`.

**Comment by maxim-sheronov** [comment]
@neikeq I found it! Thanks. I will fix it and prepare pull request

**Comment by maxim-sheronov** [comment]
@neikeq could you help me? As I describe, there is one more problem. For example enum LineTextureMode declared not in class Line2D, but binded to them. That produces wrong output api. But this I fixed by moving enum to class Line2D.

But later I found more complex problem. It is PowerState enum. It binded to _OS class. But os/power.h header is used in many other places. Ant I can't just move this enum to _OS class. What can be the best solution in this situation?

**Comment by maxim-sheronov** [comment]
Maybe I should create Power class and bind this enum to it instead of _OS

**Comment by maxim-sheronov** [comment]
I added a PR for this issue #11230

**Comment by neikeq** [comment]
We could have an alternative `VARIANT_ENUM_CAST` macro for this particular cases which are not common. e.g.:

``` C++
// Same as VARIANT_ENUM_CAST but the class_name would be _OS.PowerState instead
VARIANT_ENUM_CAST_FOR_CLASS(Power::PowerState, _OS);
```

**Comment by akien-mga** [comment]
The `Power` stuff should likely be integrated into `OS` proper, it doesn't really have to be treated differently from other OS interfaces.

**Comment by maxim-sheronov** [comment]
@akien-mga ok, I will try to change it

**Comment by maxim-sheronov** [comment]
Fixed in #11230
